lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py

```diff
@@ -322,7 +322,7 @@ def __init__(self, namespace: str, statefulset_name: str, container_name: str):
         self.namespace = namespace
         self.statefulset_name = statefulset_name
         self.container_name = container_name
-        self.client = Client()
+        self.client = Client()  # pyright: ignore
```

Comment:
can we solve whatever issue it is forcing us to add this ignore instead?

Reply:
mm, it seems to be in the upstream

Reply:
Client's signature has something like `foo: Bar = None` and pyright seems unhappy with that 'implicit' `Optional`.
We could submit a pr to the upstream to make it `foo: Optional[Bar] = None`, but I'm not sure if it's worth the effort, and if 'being compatible with pyright' is a good reason for a third-party product to accept a change.

Reply:
Correct about the issue being upstream; this is something we already did for everything else importing lightkube :)

Reply:
What is the issue? Can you paste the error pyright prints?

Reply:
```
/home/aegis/Repositories/Canonical/observability-libs/lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py:110:33 - error: "Client" is not exported from module "lightkube"
```

Pietro described the issue with it already; again, this *could* be fixed by importing the `Client` directly, but it's a lot of code changes, and I'd rather keep this PR small; just as a reminder, `mypy` wasn't checking any of this, that's why we didn't get errors :)